test/core/loader/particles-loader/particle-file.test.js

```diff
@@ -39,6 +42,54 @@ describe("Particles File", () => {
             expect(global.JSON.parse).toHaveBeenCalledWith(testJSON);
         });
 
+        test("calls the the Geom parser when a emitZone source is provided", () => {
+            const file = new ParticlesFile(mockLoader, mockFileConfig);
+            const testJSON = '{ "emitZone": { "source": { "emitZone": "mock" } } }';
+
+            file.xhrLoader = {
+                responseText: testJSON,
+            };
+            file.onProcessComplete = jest.fn();
+            file.onProcess();
+            expect(geomParse).toHaveBeenCalledWith({ emitZone: "mock" });
+        });
+
+        test("sets return value on data when a emitZone source is provided", () => {
+            const file = new ParticlesFile(mockLoader, mockFileConfig);
+            const testJSON = '{ "emitZone": { "source": { "emitZone": "mock" } } }';
+            geomParse.mockImplementation(() => "mock emit zone");
+            file.xhrLoader = {
+                responseText: testJSON,
+            };
+            file.onProcessComplete = jest.fn();
+            file.onProcess();
+            expect(file.data).toEqual({ emitZone: { source: "mock emit zone" } });
+        });
+
+        test("calls the the Geom parser when a deathZone source is provided", () => {
```

Comment:
Do we need a test to check that the Geom parser is not called when there is niether a death zone or an emit zone?

Reply:
added, thanks 👍